wasm-chain-integration/src/utils.rs

```diff
@@ -367,17 +369,30 @@ pub fn get_embedded_schema_v0(bytes: &[u8]) -> ExecResult<schema::ModuleV0> {
 }
 
 /// Get the embedded schema if it exists
-pub fn get_embedded_schema_v1(bytes: &[u8]) -> ExecResult<schema::ModuleV1> {
+/// It will attempt to use the schema in the custom section "concordium-schema"
+/// and if this is not present it will try to use the custom section
+/// "concordium-schema-v2".
+pub fn get_embedded_schema_versioned(bytes: &[u8]) -> ExecResult<schema::VersionedModule> {
     let skeleton = parse_skeleton(bytes)?;
-    let mut schema_sections = Vec::new();
+    let mut schema_v2_section = None;
+    let mut schema_versioned_section = None;
     for ucs in skeleton.custom.iter() {
         let cs = parse_custom(ucs)?;
-        if cs.name.as_ref() == "concordium-schema-v2" {
-            schema_sections.push(cs)
+        if cs.name.as_ref() == "concordium-schema" && schema_versioned_section.is_none() {
+            schema_versioned_section = Some(cs)
+        } else if cs.name.as_ref() == "concordium-schema-v2" && schema_v2_section.is_none() {
+            schema_v2_section = Some(cs)
         }
     }
-    let section =
-        schema_sections.first().ok_or_else(|| anyhow!("No schema found in the module"))?;
-    let source = &mut Cursor::new(section.contents);
-    source.get().map_err(|_| anyhow!("Failed parsing schema"))
+
+    if let Some(cs) = schema_versioned_section {
+        let module = from_bytes(&cs.contents).map_err(|_| anyhow!("Failed parsing schema"))?;
+        return Ok(module);
```

Comment:
Above you first check for `concordium-schema` but here you first check for the versioned schema.

Please make the order consistent. And document that one takes precedence over the other in the comment to the function.

Reply:
I am not sure what the problem is here, I have already commented the precedence in the doc commenr for the function and the check seems consistent to me.
Maybe you are misunderstanding my naming of variables?

Reply:
Yes, I got confused. Sorry.